src/middlewared/middlewared/utils/osc/linux/user_context.py

```diff
@@ -41,6 +44,10 @@ def set_user_context(user: str) -> None:
 
 
 def run_with_user_context(func: Callable, user: str, func_args: Optional[list] = None) -> Any:
+    try:
+        pwd.getpwnam(user)
```

Comment:
We have method that wraps around this. `user.get_user_obj` We should use this. Maybe pass the resulting dict into set_user_context.

Reply:
```
    @accepts(Dict(
        'get_user_obj',
        Str('username', default=None),
        Int('uid', default=None),
        Bool('get_groups', default=False)
    ))
```

May want to use `get_groups` option as well so that we can setresuid() into correct list of users / groups.